[Bug]: libwayland build failed ndk25

### Problem description

./protocol/wayland-client-protocol.h:5437:97: error: use of undeclared identifier 'WL_MARSHAL_FLAG_DESTROY'
                         WL_SUBCOMPOSITOR_DESTROY, NULL, wl_proxy_get_version((struct wl_proxy *) wl_subcompositor), WL_MARSHAL_FLAG_DESTROY);
                                                                                                                     ^
./protocol/wayland-client-protocol.h:5464:7: warning: implicit declaration of function 'wl_proxy_marshal_flags' is invalid in C99 [-Wimplicit-function-declaration]
        id = wl_proxy_marshal_flags((struct wl_proxy *) wl_subcompositor,
             ^
./protocol/wayland-client-protocol.h:5464:5: warning: incompatible integer to pointer conversion assigning to 'struct wl_proxy *' from 'int' [-Wint-conversion]
        id = wl_proxy_marshal_flags((struct wl_proxy *) wl_subcompositor,
           ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./protocol/wayland-client-protocol.h:5545:2: warning: implicit declaration of function 'wl_proxy_marshal_flags' is invalid in C99 [-Wimplicit-function-declaration]
        wl_proxy_marshal_flags((struct wl_proxy *) wl_subsurface,
        ^
./protocol/wayland-client-protocol.h:5546:91: error: use of undeclared identifier 'WL_MARSHAL_FLAG_DESTROY'
                         WL_SUBSURFACE_DESTROY, NULL, wl_proxy_get_version((struct wl_proxy *) wl_subsurface), WL_MARSHAL_FLAG_DESTROY);
                                                                                                               ^
./protocol/wayland-client-protocol.h:5572:2: warning: implicit declaration of function 'wl_proxy_marshal_flags' is invalid in C99 [-Wimplicit-function-declaration]
        wl_proxy_marshal_flags((struct wl_proxy *) wl_subsurface,
        ^
./protocol/wayland-client-protocol.h:5598:2: warning: implicit declaration of function 'wl_proxy_marshal_flags' is invalid in C99 [-Wimplicit-function-declaration]
        wl_proxy_marshal_flags((struct wl_proxy *) wl_subsurface,
        ^
./protocol/wayland-client-protocol.h:5611:2: warning: implicit declaration of function 'wl_proxy_marshal_flags' is invalid in C99 [-Wimplicit-function-declaration]
        wl_proxy_marshal_flags((struct wl_proxy *) wl_subsurface,
        ^
./protocol/wayland-client-protocol.h:5635:2: warning: implicit declaration of function 'wl_proxy_marshal_flags' is invalid in C99 [-Wimplicit-function-declaration]
        wl_proxy_marshal_flags((struct wl_proxy *) wl_subsurface,
        ^
./protocol/wayland-client-protocol.h:5665:2: warning: implicit declaration of function 'wl_proxy_marshal_flags' is invalid in C99 [-Wimplicit-function-declaration]
        wl_proxy_marshal_flags((struct wl_proxy *) wl_subsurface,
        ^
79 warnings and 14 errors generated.
make[2]: *** [Makefile:1793: src/libwayland_client_la-wayland-client.lo] Error 1
make[2]: Leaving directory '/home/builder/.termux-build/libwayland/build'
make[1]: *** [Makefile:2036: all-recursive] Error 1
make[1]: Leaving directory '/home/builder/.termux-build/libwayland/build'
make: *** [Makefile:1238: all] Error 2
builder@3734a7c6abeb:~/termux-packages$


[libwayland.txt](https://github.com/termux/termux-packages/files/9333792/libwayland.txt)


### What steps will reproduce the bug?

build

### What is the expected behavior?

_No response_

### System information

termux-info:

```

```

--- a/x11-packages/libwayland/build.sh
+++ b/x11-packages/libwayland/build.sh
@@ -3,14 +3,30 @@ TERMUX_PKG_DESCRIPTION="Wayland protocol library"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION=1.17.0
-TERMUX_PKG_REVISION=9
+TERMUX_PKG_REVISION=10
 TERMUX_PKG_SRCURL=https://wayland.freedesktop.org/releases/wayland-${TERMUX_PKG_VERSION}.tar.xz
 TERMUX_PKG_SHA256=72aa11b8ac6e22f4777302c9251e8fec7655dc22f9d94ee676c6b276f95f91a4
 TERMUX_PKG_DEPENDS="libandroid-support, libffi, libxml2, libexpat"
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--disable-documentation --with-host-scanner"
+TERMUX_PKG_HOSTBUILD=true
+TERMUX_PKG_EXTRA_HOSTBUILD_CONFIGURE_ARGS="
+--disable-libraries
+--disable-documentation
+--disable-dtd-validation
+--prefix=$TERMUX_PREFIX/opt/$TERMUX_PKG_NAME/cross
+"
+
+termux_step_host_build() {
+	"$TERMUX_PKG_SRCDIR/configure" ${TERMUX_PKG_EXTRA_HOSTBUILD_CONFIGURE_ARGS}
+	make -j "$TERMUX_MAKE_PROCESSES"
+	make install
+}
 
 termux_step_pre_configure() {
 	autoreconf -fi
+
+	_HOST_PREFIX=$TERMUX_PREFIX/opt/$TERMUX_PKG_NAME/cross
+	export PATH=${_HOST_PREFIX}/bin:$PATH
 }
 
 termux_step_post_make_install() {

--- a/x11-packages/libwayland/libwayland-cross-scanner.subpackage.sh
+++ b/x11-packages/libwayland/libwayland-cross-scanner.subpackage.sh
@@ -1,0 +1,4 @@
+TERMUX_SUBPKG_DESCRIPTION="wayland-scanner for host (NOT for Termux)"
+TERMUX_SUBPKG_INCLUDE="
+opt/$TERMUX_PKG_NAME/cross/*
+"

--- a/x11-packages/xwayland/build.sh
+++ b/x11-packages/xwayland/build.sh
@@ -3,12 +3,15 @@ TERMUX_PKG_DESCRIPTION="Wayland X11 server"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION=1.20.5
-TERMUX_PKG_REVISION=14
+TERMUX_PKG_REVISION=15
 TERMUX_PKG_SRCURL=https://xorg.freedesktop.org/releases/individual/xserver/xorg-server-${TERMUX_PKG_VERSION}.tar.bz2
 TERMUX_PKG_SHA256=a81d8243f37e75a03d4f8c55f96d0bc25802be6ec45c3bfa5cb614c6d01bac9d
 TERMUX_PKG_DEPENDS="libandroid-shmem, libdrm, libpciaccess, libpixman, libx11, libxau, libxfont2, libxinerama, libxkbfile, libxshmfence, mesa, openssl, xkeyboard-config, xorg-xkbcomp, libwayland, libwayland-protocols, libepoxy"
+TERMUX_PKG_BUILD_DEPENDS="libwayland-cross-scanner"
 
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+ac_cv_path_RAWCPP=/usr/bin/cpp
+ac_cv_path_WAYLAND_SCANNER=$TERMUX_PREFIX/opt/libwayland/cross/bin/wayland-scanner
 --enable-composite
 --enable-mitshm
 --enable-xres

